### ISSUE TITLE
Support for self-signed certificates

### ISSUE THREAD

**Issue by prog-amateur2** [readrops/Readrops#48]
Hello, when I try to sign in Nexctcloud News, I have this following error message : 
```
java.security.cert.CertpathValidatorException: 
Trust Anchor for certification path not found
```

Do you have any idea why this happen ? I know that I have a self certificate but in Nextcloud News app, I can click on trust while here, there is only error message. Thank you very much.

**Comment by Shinokuni** [comment]
Hello, Readrops currently doesn't have any kind of support for self-signed certificates.

It is possible to add, but it will require some work as I will need a self-signed instance for testing purpose.

**Comment by prog-amateur2** [comment]
Hello, thank you for your quick reply. You can take your time, I was ready to close this issue, but I see a label `enhancement`. Shall I close this issue or do you prefer to keep it open for your follow up ? Thank you

**Comment by Shinokuni** [comment]
Keep it open, I will use it as follow up.

**Comment by chris-si** [comment]
Hello, I'm also interested in this enhancement since my FreshRSS instance uses a self signed certificate and I get the same error message. Before I found this issue I tried to import my ca certificate into the android certificate store but unfortunately this didn't help getting the app to work with my FreshRSS instance.

**Comment by yps** [comment]
Any update on this? This is quite important for selfhosted environments.

**Comment by Shinokuni** [comment]
Support for self-signed certificates was added in 489f52b13a0fcd91a623a10a36245fad8d721f71. It will be available in 1.4.

**Comment by gbakeman** [comment]
Any chance we can have an update on what the plan is for 1.4? I'm currently blocked from the using the app because I maintain my own internal CA. Android is OK with trusting it as a user cert for most apps so it would be great to have this feature. I'd be happy to run a test build.
